src/main/java/core/basesyntax/operations/PurchaseOperation.java

```diff
@@ -0,0 +1,12 @@
+package core.basesyntax.operations;
+
+import core.basesyntax.FruitTransaction;
+import java.util.Map;
+
+public class PurchaseOperation implements OperationHandler {
+    @Override
+    public void apply(Map<String, Integer> storage, FruitTransaction transaction) {
+        storage.put(transaction.getFruit(),
+                storage.getOrDefault(transaction.getFruit(), 0) - transaction.getQuantity());
```

Comment:
Before updating the storage, you need to check if the resulting balance after the purchase will be positive. If the balance is negative, you should throw a RuntimeException as per the checklist requirement 'Handling Purchase operation'.